### PR TITLE
Bug 2004510: ztp: Add authentication retries to ztp container

### PR DESCRIPTION
**Pull request by imiller0**
Add retry logic to cluster API authentication and fast-fail if
authentication fails after 25 retries.

Signed-off-by: Ian Miller <imiller@redhat.com>

### CODE REVIEW

File: ztp/resource-generator/src/common.sh [hunk 1]
```diff
@@ -1,11 +1,28 @@
...
-export CACERT=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt
-export NAMESPACE=$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)
+export CHECK_RC=1
+for idx in {25..1} ; do
+    export APISERVER=https://kubernetes.default.svc:443
+    export TOKEN=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
+    export CACERT=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+    export NAMESPACE=$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)
 
-oc login $APISERVER --token=$TOKEN --certificate-authority=$CACERT &> /dev/null
```
Comment (lack) [inline ztp/resource-generator/src/common.sh:8]: Nice job getting the token off the cmdline!  Security approves :)

File: ztp/resource-generator/src/get-pre-sync-rv.sh [hunk 2]
```diff
@@ -1,5 +1,8 @@
 #!/bin/bash
 
+# Log in and setup environment
+. $(dirname "$0")/common.sh $1
```
Comment (lack) [inline ztp/resource-generator/src/get-pre-sync-rv.sh:4]: If this is the way it's used, perhaps it's better to put the code in a function, as opposed to the work being done as a side-effect of sourcing it?

```
. $(dirname "$0")/common.sh

# Login and setup environment
waitForLogin $1
```

File: ztp/resource-generator/src/post-sync-entrypoint.sh [hunk 3]
```diff
@@ -2,4 +2,4 @@
 
 . $(dirname "$0")/common.sh $1
```
Comment (lack) [inline ztp/resource-generator/src/post-sync-entrypoint.sh:3]: What is `$1`?  Can we give it a name?

File: ztp/resource-generator/src/get-pre-sync-rv.sh (hunk 2 above)
Comment (lack) [reply ztp/resource-generator/src/get-pre-sync-rv.sh:4]: This approach would also lend itself to eventually adding a unit test around the `waitForLogin` function or whatever you want to call it.

Comment (imiller0) [reply ztp/resource-generator/src/get-pre-sync-rv.sh]: Agreed. Reworked to be a function. Side-effects from loading aren't nice :)

**Review by lack — APPROVED** [review]

**Pull request merged**